src/components/VideoDetails/VideoDetails.component.jsx

```diff
@@ -16,9 +19,36 @@ import {
 } from './VideoDetails.styles';
 
 const VideoDetails = ({ videoId, description, title, videos }) => {
+  const [isFavorite, setIsFavorite] = useState(false);
+  const { authenticated } = useAuth();
   const { state } = useGlobalState();
   const { isThemeLight } = state;
 
+  useEffect(() => {
+    const myFavs = storage.get(FAV_STORAGE_KEY);
+    if (myFavs) {
+      setIsFavorite(Boolean(myFavs[videoId]));
+    } else {
+      setIsFavorite(false);
+    }
+  }, [videoId]);
+
+  const getCurrentVideo = () => {
+    return videos.filter((video) => videoId === video.id.videoId);
```

Comment:
If for some reason id is not present or of a different type, you could get a breaking error here, I suggest to use defaults instead, like
```suggestion
    return videos.filter(({ id = {} } = {}) => videoId === id.videoId);
```